eos-fix-home-dir-permissions

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -e
+
+# Prior to 3.1.6, home directory permissions were created with dir mode 755.
+# This is now fixed, but on upgrade from an earlier version to 3.1.6,
+# let's perform a one-time clean up of any existing home dirs.
+
+stamp_file=/var/eos-dir-mode-700
```

Comment:
Hmm... stampy stampy. I'd go as far as to say /var/lib/misc because /var/lib is arranged under a hierachy too.

Not sure if it's a red herring seeing as this is actually quite neat with a ConditionPathExists, but can you not check the permissions of /home/shared as the only state you need to determine whether this job needs to run? I can't immediately see a way to hook that to systemd so it would be "strictly worse" unfortunately, in that the job would need to run to decide if it needed to run or not, so would add to the default boot path.

(I do wonder if we could do better with systemd units which need to be run when upgrading from a version before X, and keep one file with the version in a fixed location and do a monotonic check, trigger an events which other systemd tasks can bind to, then update the file... hmm...)